lib/process/child-pool.js

```diff
@@ -41,36 +41,50 @@ const convertExecArgv = function(execArgv) {
 
 ChildPool.prototype.retain = function(processFile) {
   const _this = this;
-  let child = _this.getFree(processFile).pop();
+  const child = _this.getFree(processFile).pop();
 
   if (child) {
     _this.retained[child.pid] = child;
     return Promise.resolve(child);
   }
 
-  return convertExecArgv(process.execArgv).then(execArgv => {
-    child = fork(path.join(__dirname, './master.js'), {
-      execArgv
-    });
-    child.processFile = processFile;
-
+  return this.forkChild(processFile).then(child => {
     _this.retained[child.pid] = child;
 
-    child.on('exit', _this.remove.bind(_this, child));
-
     return initChild(child, child.processFile).then(() => {
       return child;
     });
   });
 };
 
+ChildPool.prototype.addFreeChild = function(processFile) {
+  return this.forkChild(processFile).then(child => {
+    this.getFree(child.processFile).push(child);
+    return initChild(child, child.processFile).then(() => child);
+    })
+};
+
+ChildPool.prototype.forkChild =  function(processFile) {
+  const _this = this;
```

Comment:
> I think it would be better if the child pool can be passed as an optional argument when instantiating the queue instead of this new method "initChildPool" which is not so easy to understand (at least for me).

@manast yeah i definitely see your point I created this function because I was under the assumption that the fork process was asynchronous, but after double checking I see that it is a asynchronous and can make changes accordingly

Reply:
@manast  I reviewed the code and the reason I didn't add the fork process code into the constructor is because of the [initChild](https://github.com/OptimalBits/bull/blob/27ca17bdcbe82f521a92bc5c4d875a9f5375df3e/lib/process/child-pool.js#L108) function in child pool class initializes the childPool to process jobs. I'm thinking maybe the next logical place to put it would be inside of the `queue.process` hander